src/Symfony/Component/DependencyInjection/Tests/EnvVarProcessorTest.php

```diff
@@ -0,0 +1,182 @@
+
+        $processor = new EnvVarProcessor($container);
+
+        $result = $processor->getEnv('string', 'foo', function () {
+            throw new LogicException('Shouldnt be called');
+        });
+
+        $this->assertSame('hello', $result);
+    }
+
+    public function testGetEnvBool()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('bool', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return '1';
+        });
+
+        $this->assertTrue($result);
+    }
+
+    public function testGetEnvInt()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('int', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return '1';
+        });
+
+        $this->assertSame(1, $result);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Non-numeric env var
+     */
+    public function testGetEnvIntInvalid()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $processor->getEnv('int', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return 'bar';
+        });
+    }
+
+    public function testGetEnvFloat()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('float', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return '1.1';
+        });
+
+        $this->assertSame(1.1, $result);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Non-numeric env var
+     */
+    public function testGetEnvFloatInvalid()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $processor->getEnv('float', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return 'bar';
+        });
+    }
+
+    public function testGetEnvConst()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $result = $processor->getEnv('const', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return 'Symfony\Component\DependencyInjection\Tests\EnvVarProcessorTest::TEST_CONST';
+        });
+
+        $this->assertSame(self::TEST_CONST, $result);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage undefined constant
+     */
+    public function testGetEnvConstInvalid()
+    {
+        $processor = new EnvVarProcessor(new Container());
+
+        $processor->getEnv('const', 'foo', function ($name) {
+            $this->assertSame('foo', $name);
+
+            return 'Symfony\Component\DependencyInjection\Tests\EnvVarProcessorTest::TEST_CONST_OTHER';
```

Comment:
I'd suggest to use a name here that tells the reader that you've been using an undefined constant by design here, e.g. `TEST_CONST_INVALID` or `TEST_CONST_UNDEFINED`.